### PR TITLE
[Storage] Directory Pause and Resume Tests with directory resume fixes

### LINKED ISSUE #35435: [storage] [datamovement] Single Download, Single Copy and Directory Pause and Resume Tests
### Library name

Azure.Storage.DataMovement; Azure.Storage.DataMovement.Blobs

### Please describe the feature.

See title.

### PR DESCRIPTION
**Pull request by amnguye**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/35435

This PR includes tests for Pause and Resume specific to Directory Pause and Resume, fixes to any bugs found during the development of pause and resume of directory transfers, and also ensuring that all the chunk tasks are running on their own task thread and not occupying the parent thread.

Added in each transfer job part is the property of whether or not it was the last job part which is recorded in the job part plan file. This ensures whether or not enumeration has finished. In the future it is discussed to remove the value, but as of the current format of the job part plan file it will be kept this way.

Other minor changes unrelated to the directory pause and resume
- S minor change to an exception message (overwrite was mispelled)
- Enabling download error tests with fixed error message

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/tests/BlockBlobClientTests.cs [hunk 1]
```diff
@@ -2788,7 +2788,7 @@ public async Task SyncUploadFromUriAsync_Error()
 
         [RecordedTest]
         [ServiceVersion(Min = BlobClientOptions.ServiceVersion.V2020_04_08)]
-        public async Task SyncUploadFromUriAsync_OverwiteSourceBlobProperties()
+        public async Task SyncUploadFromUriAsync_overwriteSourceBlobProperties()
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Blobs/tests/BlockBlobClientTests.cs:2791]: TODO: renaming this back to correct camel case. I think this got caught in my mass rename of overwite to overwrite

File: sdk/storage/Azure.Storage.DataMovement/tests/PauseResumeTransferTests.cs [hunk 2]
```diff
@@ -162,7 +227,7 @@ public enum TransferType
         [TestCase(TransferType.Download)]
         [TestCase(TransferType.AsyncCopy)]
         [TestCase(TransferType.SyncCopy)]
-        [Ignore("https://github.com/Azure/azure-sdk-for-net/issues/35439")]
+        //[Ignore("https://github.com/Azure/azure-sdk-for-net/issues/35439")]
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.DataMovement/tests/PauseResumeTransferTests.cs:230]: TODO: put back the ignore as these tests can be flakey.

File: sdk/storage/Azure.Storage.DataMovement/src/UriToStreamTransferJob.cs [hunk 3]
```diff
@@ -71,49 +72,159 @@ public override async IAsyncEnumerable<JobPartInternal> ProcessJobToJobPartAsync
...
+                    {
+                        yield return jobPartInternal;
                     }
                 }
             }
             _enumerationComplete = true;
             await OnEnumerationComplete().ConfigureAwait(false);
         }
+
+        private async IAsyncEnumerable<JobPartInternal> GetStorageResourcesAsync()
```
Comment (jalauzon-msft) [inline sdk/storage/Azure.Storage.DataMovement/src/UriToStreamTransferJob.cs:130]: This function exists on each job type, can we move it to base class or a separate helper somehow?

Comment (jalauzon-msft) [reply sdk/storage/Azure.Storage.DataMovement/src/UriToStreamTransferJob.cs:130]: Sure, no problem. Just probably mark something in that issue so we don't forget :).

**Review by jalauzon-msft — APPROVED** [review]

**Pull request merged**